Source/CarthageKit/Xcode.swift

```diff
@@ -1390,64 +1406,69 @@ private func stripArchitecture(frameworkURL: NSURL, _ architecture: String) -> S
 		.then(.empty)
 }
 
-/// Returns a signal of all architectures present in a given framework.
-public func architecturesInFramework(frameworkURL: NSURL) -> SignalProducer<String, CarthageError> {
+/// Returns a signal of all architectures present in a given package.
+public func architecturesInPackage(packageURL: NSURL) -> SignalProducer<String, CarthageError> {
 	return SignalProducer.attempt { () -> Result<NSURL, CarthageError> in
-			return binaryURL(frameworkURL)
+			return binaryURL(packageURL)
+		}
+		.flatMap(.Merge) { binaryURL in
+			return architecturesInBinary(binaryURL)
 		}
-		.flatMap(.Merge) { binaryURL -> SignalProducer<String, CarthageError> in
-			let lipoTask = Task("/usr/bin/xcrun", arguments: [ "lipo", "-info", binaryURL.path!])
+}
 
-			return launchTask(lipoTask)
-				.ignoreTaskData()
-				.mapError(CarthageError.TaskError)
-				.map { NSString(data: $0, encoding: NSUTF8StringEncoding) ?? "" }
-				.flatMap(.Merge) { output -> SignalProducer<String, CarthageError> in
-					let characterSet = NSMutableCharacterSet.alphanumericCharacterSet()
-					characterSet.addCharactersInString(" _-")
-
-					let scanner = NSScanner(string: output as String)
-
-					if scanner.scanString("Architectures in the fat file:", intoString: nil) {
-						// The output of "lipo -info PathToBinary" for fat files
-						// looks roughly like so:
-						//
-						//     Architectures in the fat file: PathToBinary are: armv7 arm64
-						//
-						var architectures: NSString?
-
-						scanner.scanString(binaryURL.path!, intoString: nil)
-						scanner.scanString("are:", intoString: nil)
-						scanner.scanCharactersFromSet(characterSet, intoString: &architectures)
-
-						let components = architectures?
-							.componentsSeparatedByString(" ")
-							.filter { !$0.isEmpty }
-
-						if let components = components {
-							return SignalProducer(values: components)
-						}
-					}
+/// Returns a signal of all architectures present in a given binary.
+public func architecturesInBinary(binaryURL: NSURL) -> SignalProducer<String, CarthageError> {
```

Comment:
Why did you split this out from `architecturesInPackage`? It looks like it's still only called there.